webclient/arch_z/tests/test_selenium.py

```diff
@@ -0,0 +1,1326 @@
+import time
+import datetime
+import logging
+import unittest
+import base64
+
+from django.conf import settings
+from selenium.webdriver import ActionChains
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+from core.tests.test_selenium import BaseSeleniumTestClass, Wait_for_page_load
+from arch_z.models import Akce,ExterniOdkaz
+from dokument.models import DokumentCast
+from dj.models import DokumentacniJednotka
+from komponenta.models import Komponenta
+from pian.models import Pian
+from nalez.models import NalezObjekt, NalezPredmet
+from django.utils.translation import gettext as _
+from selenium.common.exceptions import *
+logger = logging.getLogger("tests")
+
+
+@unittest.skipIf(settings.SKIP_SELENIUM_TESTS, "Skipping Selenium tests")
+class AkceProjektoveAkce(BaseSeleniumTestClass):   
+    def go_to_Projekty_vyper(self):
+        self.ElementClick(By.CSS_SELECTOR, ".card:nth-child(1) .btn")
+        self.ElementClick(By.LINK_TEXT, _("templates.baseLogedIn.sidebar.projekty.vybratProjekty")) 
+
+    def  draw_polygon(self):       
+        self.wait(1)
+        self.driver.execute_script('''map.setZoom(16); return map.getZoom();''')
+        self.wait(2)  
+        self.ElementClick(By.LINK_TEXT, _("mapa.EditAddPolygon"))
+        self.clickAt(self.driver.find_element(By.ID, "djMap"),0,0)
+        self.wait(0.2) 
+        self.clickAt(self.driver.find_element(By.ID, "djMap"),0,20)
+        self.wait(0.2) 
+        self.clickAt(self.driver.find_element(By.ID, "djMap"),20,20)
+        self.wait(0.2) 
+        self.clickAt(self.driver.find_element(By.ID, "djMap"),20,0)
+        self.wait(0.2) 
+        self.ElementClick(By.LINK_TEXT, _("mapa.EditFinishText"))
+        self.wait(0.2) 
+    
+    def test_024_pridani_dokumentacni_jednotky_p_001(self):
+        #Scenar_24 Přidání dokumentační jednotky celek akce (pozitivní scénář 1)
+        logger.info("AkceProjektoveAkce.test_024_pridani_dokumentacni_jednotky_p_001.start")
+
+        self.login()
+        self.go_to_Projekty_vyper()
+        
+        count_old=DokumentacniJednotka.objects.filter(archeologicky_zaznam__ident_cely="C-202307816A").count()
+        
+        self.ElementClick(By.CSS_SELECTOR, ".mt-1")
```

Comment:
Obecně bych doporučoval přidání nějaké vlastní třídy nebo ID HTML prvku do formuláře a poté řízení kliknutí podle nich. Ty bootstrap třídy můžou být v rámci úprav frontendu přidávány a odebírány a zbytečně to pak způsobí, že budou testy padat. Navíc z popisu není jasné, na Selenium kliká.

Reply:
Určitě to vezmu v potaz, až budu dělat další testy, ale teď to tak nechám a opravím to až dojde k nějakým změnám.